### PR TITLE
Merge sources into one repo

### PR DESCRIPTION
**Pull request by andrueastman**
Fixes https://github.com/microsoft/kiota-abstractions-python/issues/302
Fixes https://github.com/microsoft/kiota-abstractions-python/issues/174

Todo

- [x] Resolved mypy issues via https://github.com/microsoft/kiota-abstractions-python/pull/331
- [x] Unify versioning across projects via https://github.com/microsoft/kiota-abstractions-python/pull/344
- [x] Sync changes for https://github.com/microsoft/kiota-serialization-json-python/pull/369 via https://github.com/microsoft/kiota-abstractions-python/pull/345
- [x] Validate workflows

### CODE REVIEW

File: packages/authentication/azure/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Microsoft Kiota Authentication Azure Library for Python
+[![PyPI version](https://badge.fury.io/py/microsoft-kiota-authentication-azure.svg)](https://badge.fury.io/py/microsoft-kiota-authentication-azure)
+[![CI Actions Status](https://github.com/microsoft/kiota-authentication-azure-python/actions/workflows/build.yml/badge.svg?branch=main)](https://github.com/microsoft/kiota-authentication-azure-python/actions)
```
Comment (baywet) [inline packages/authentication/azure/README.md:3]: badge needs to be updated

File: packages/http/httpx/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Microsoft Kiota HTTP library
+[![PyPI version](https://badge.fury.io/py/microsoft-kiota-http.svg)](https://badge.fury.io/py/microsoft-kiota-http)
+[![CI Actions Status](https://github.com/microsoft/kiota-http-python/actions/workflows/build.yml/badge.svg?branch=main)](https://github.com/microsoft/kiota-http-python/actions)
```
Comment (baywet) [inline packages/http/httpx/README.md:3]: badge needs to be updated

**Review by baywet — COMMENTED** [review]
started reviewing the PR but stopped midway through since I realized #331 still needs to be merged in this branch anyway

**Comment by andrueastman** [comment]
> started reviewing the PR but stopped midway through since I realized https://github.com/microsoft/kiota-abstractions-python/pull/331 still needs to be merged in this branch anyway

Thanks @baywet, Just to set expectations here, once we close https://github.com/microsoft/kiota-abstractions-python/pull/331, I'll need to make some updates to do the following before this is fully ready for review.

- Cleanup readmes and related docs
- Validate workflows
- Align versioning

**Review by baywet — DISMISSED** [review]

**Comment by baywet** [comment]
@andrueastman the policy definition probably needs to be updated here

**Comment by andrueastman** [comment]
agreed.
I'll open a sepate PR targeting main in order to close https://github.com/microsoft/kiota-abstractions-python/issues/327 before we merge this.

**Review by musale — APPROVED** [review]
Good stuff @andrueastman!

**Comment by baywet** [comment]
this PR also needs to be ported https://github.com/microsoft/kiota-http-python/pull/434

**Comment by andrueastman** [comment]
- Sync latest sources at https://github.com/microsoft/kiota-abstractions-python/pull/351 
- Resolving policy in main branch before merge at https://github.com/microsoft/kiota-abstractions-python/pull/350

**Review by baywet — APPROVED** [review]

**Review by musale — APPROVED** [review]

**Pull request merged**